addons/mail/controllers/discuss/search.py

```diff
@@ -0,0 +1,33 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+from odoo.http import request
+from odoo.addons.mail.models.discuss.mail_guest import add_guest_to_context
+from odoo.addons.mail.tools.discuss import Store
+
+
+class SearchController(http.Controller):
+    @http.route("/discuss/search", methods=["POST"], type="json", auth="public")
+    @add_guest_to_context
+    def search(self, term, category_id=None, limit=8):
+        return self.get_search_store(term, category_id, limit).get_result()
+
+    def get_search_store(self, term, category_id, limit):
+        store = Store()
```

Comment:
It's not a good guideline to have a sub-method creating the store, it should always be provided from the main method (the one calling get_result) in order to be able to combine and flatten results in more complex flows.

Reply:
In this case I don't see the point of `get_search_store` before a separate method, it looks called only once and not overriden?